app/Filament/Server/Resources/FileResource/Pages/ListFiles.php

```diff
@@ -502,6 +502,7 @@ protected function getHeaderActions(): array
                                         ->storeFiles(false)
                                         ->previewable(false)
                                         ->preserveFilenames()
+                                        ->maxSize($server->node->upload_size * config('panel.use_binary_prefix') ? 1024 : 1000)
```

Comment:
As far as I can see `maxSize` always expects KiB (despite the filamentphp docs calling it KB). So we need to do some math when using decimal prefix.